cmd/updater/update.sh

```diff
@@ -330,20 +340,67 @@ function validate_update() {
     return 0
 }
 
+function check_service() {
```

Comment:
there's nothing in common between the "user" and "system" branches of this function. It might be clearer as two different functions check_service_user() and check_service_system()

Reply:
I initially had them as two separate functions, but then the only bits there were different between them were `--user` and it seemed better to combine them.